### PR TITLE
Refactor dapps repository - Closes #1407

### LINKED ISSUE #1407: Refactor dapps db repository
Parent: #1376 

This is specifically for repository `dapps`.

### PR DESCRIPTION
**Pull request by vitaly-t**
Closes #1407

### CODE REVIEW

**Comment by vitaly-t** [comment]
I'm currently mystified by the failure in Jenkins: https://jenkins.lisk.io/job/lisk-core/view/change-requests/job/PR-1408/2/console

Can't see how, if at all, it is related to the PR. Quite possibly, not.

File: db/repos/dapps.js [hunk 1]
```diff
@@ -22,87 +22,84 @@ var PQ = require('pg-promise').ParameterizedQuery;
...
+		this.sortFields = ['name'];
+	}
+
+	/**
+	 * Counts dapps by transaction id
+	 * @param {string} id
+	 * @return {Promise<number>}
+	 */
+	countByTransactionId (id) {
+		return this.db.one(sql.countByTransactionId, id, a => +a.count);
```
Comment (nazarhussain) [inline db/repos/dapps.js]: The usage of the code is expecting a row object with a count attribute.  This change will break the code. 

https://github.com/LiskHQ/lisk/blob/cc2e8998efe14b428316ae8affef135bbc6e0658/logic/inTransfer.js#L83-L85

I believe the change you did is good, but you have to update the usage of this repository method as well.

File: db/repos/dapps.js [hunk 2]
```diff
@@ -22,87 +22,84 @@ var PQ = require('pg-promise').ParameterizedQuery;
...
 
-var DappsSql = {
-	countByTransactionId: new PQ('SELECT COUNT(*)::int AS "count" FROM dapps WHERE "transactionId" = $1'),
+	/**
+	 * Counts dapps by out transfer transaction id
+	 * @param {string} id
+	 * @return {Promise<number>}
+	 */
+	countByOutTransactionId (id) {
+		return this.db.one(sql.countByOutTransactionId, id, a => +a.count);
```
Comment (nazarhussain) [inline db/repos/dapps.js]: Same here. 

https://github.com/LiskHQ/lisk/blob/cc2e8998efe14b428316ae8affef135bbc6e0658/logic/outTransfer.js#L104-L106

File: db/repos/dapps.js (hunk 1 above)
Comment (vitaly-t) [reply db/repos/dapps.js]: I have updated all usage cases as well ;)

File: db/repos/dapps.js (hunk 2 above)
Comment (vitaly-t) [reply db/repos/dapps.js]: I have updated all usage cases as well ;)

**Pull request merged**